MB-34817: Prototype for multi_match query

abhinavdangeti opened the pull request:
Directed towards addressing: https://github.com/blevesearch/bleve/issues/1249

abhinavdangeti commented:
Hey @mschoch Is this worth considering?

mschoch commented:
Two thoughts:

1.  There seems to be some complex corners to the code, like you can specify fields or analyzer or both, or neither, and there is some behavior defined for all of these.  Can we have a short call where you walk through these?

2.  ES has a multi-match query that looks similar, how similar is it?

abhinavdangeti commented:
Sounds good. Let me do a quick recap on this myself. I had looked into ES's approach on this, but I'll need a refresh on that too since it's been a while, so I'll do that before we chat.

abhinavdangeti commented:
Hey @mschoch , I've updated this commit a bit, and updated the description identifying what can be achieved using the multi_match query.
It is similar to ES in some ways. Let me know if we can chat on this on how we can do better?
